pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -511,6 +532,8 @@ class TestAnalytics extends AnalyticsImpl {
       userProperty: userProperty,
     );
 
+    if (_enableAsserts) checkBody(body);
```

Comment:
I don't understand the reasoning behind adding a separate flag when the VM already has a flag that serves the same purpose.

You could just as easily have removed the field `_enableAsserts`, changed `checkBody` to always return `true`, and written
```
assert(validBody(body));
```
It wouldn't need a message because it will always explicitly throw an exception when the `body` isn't valid.

Anyway, that seems much cleaner, and means that
- anyone can enable the checks when there's a need for it without having to write code that passes in the flag, and
- we can't ever accidentally leave the assert enabled in production code.

Reply:
Hm that's interesting, is this also the same behavior for the flutter tool? I was only made aware of this flag recently and didn't think that it was implemented for the flutter cli.

If i'm wrong, I'm happy to refactor this PR

Reply:
How we use this new flag will need to be discussed, the purpose of this PR was to have the code ready to be used in what we decide as a team.

But to continue using the `--enable-asserts` flag from dart, we could set this flag in the `Analytics` constructor to true if that flag was detected when run through the dart cli command... and on the flutter side, we can determine another approach since we don't have `--enable-asserts` for the flutter cli